ec2_vpc_nacl and ec2_vpc_nacl_info migrate to AnsibleAWSModule and add tests

tremble opened the pull request:
##### SUMMARY

Attempt to revive #37794

- Update ec2_vpc_nacl and ec2_vpc_nacl_info to use AnsibleAWSModule
- Update ec2_vpc_nacl and ec2_vpc_nacl_info docs to fix sanity test issues
- Add Integration tests for ec2_vpc_nacl and ec2_vpc_nacl_info
- Add missing AWS permissions

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

test/sanity/ignore.txt
hacking/aws_config/testing_policies/compute-policy.json
hacking/aws_config/testing_policies/network-policy.json
lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py
lib/ansible/modules/cloud/amazon/ec2_vpc_nacl_info.py

##### ADDITIONAL INFORMATION

Mostly test suite work

s-hertel committed:
ec2_vpc_nacl: Add integration tests

tremble committed:
ec2_vpc_nacl: Migrate tests to use module_defaults

tremble committed:
Move EC2 networking objects into network-policy.json

tremble committed:
ec2_vpc_nacl: (integration tests) Add missing AWS permissions

tremble committed:
ec2_vpc_nacl: (integration tests) Update tests for ipv6 support

tremble committed:
ec2_vpc_nacl: Migrate to AnsibleAWSModule

tremble committed:
Fix sanity tests for ec2_vpc_nacl and ec2_vpc_nacl_info

tremble committed:
ec2_vpc_nacl_info: Migrate to AnsibleAWSModule

tremble committed:
ec2_vpc_nacl_info: (integration tests) Rename from ec2_vpc_nacl_facts to ec2_vpc_nacl_info and add a test using a filter (by tag)

tremble committed:
Pick availability zones dynamically

Rather than assuming that AZa and AZb always exist (they don't), query to find out which AZs we have available first

jillr commented on test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml:
```diff
@@ -0,0 +1,170 @@
+        that:
+          - nacl.nacl_id
+
+    - name: remove and add an entry
```
For clarity, did you intend to also add maybe an egress rule? Or should this task just be named `remove an entry`?
Also, what errors are anticipated with the `ignore_errors` on this task?

tremble replied:
See the FIXME comments, IPv6 support is currently missing from ec2_vpc_nacl (but there in ec2_vpc_nacl_info)

 #50004 added support but needs a little work (and looks like it's abandoned)
(edited to fix PR link)

tremble replied:
(oh and the name's a hangup from @s-hertel's original work)

jillr commented on test/integration/targets/ec2_vpc_nacl/tasks/subnet_ids.yml:
```diff
@@ -0,0 +1,138 @@
+      - nacl.nacl_id.startswith('acl-')
+
+- set_fact:
+    nacl_id: "{{ nacl.nacl_id }}"
```
It does not appear this is being used anywhere.

tremble replied:
I'd intended to add a test that we were actually getting the same NACL back and not doing something screwy like deleting and recreating

jillr reviewed (commented):
Couple small queries, otherwise looks good.

tremble committed:
Test that the NACLs we get back are actually the *saml* NACL rather than duplicates/delete remove

tremble committed:
Cleanup IPv6 tests a little.

Note: IPv6 support for ec2_vpc_nacl not complete yet.

This provides the initial framework, and should ensure things don't start exploding when support is added.

tremble committed:
Removing subnets by name from a NACL *is* now supported

tremble committed:
Fix ec2_vpc_nacl return documentation
